src/struct.c

```diff
@@ -100,8 +100,12 @@ void semanticTypeInfo(Scope *sc, Type *t)
 
             // If the struct is in a non-root module, run semantic3 to get
             // correct symbols for the member function.
-            // Note that, all instantiated symbols will run semantic3.
-            if (sd->inNonRoot())
+            if (TemplateInstance *ti = sd->isInstantiated())
+            {
+                if (!ti->minst || !ti->minst->isRoot())
```

Comment:
I don't recall the reason...
Sure, when `ti` is speculative, doing `Module::addDeferredSemantic3(sd)` would be redundant, because it's already done in `TemplateInstance::appendToModuleMember`.
I'll update it.